test/test_genome.py

```diff
@@ -404,6 +404,30 @@ def test_mutate_output_region(rng_seed):
         ID_NON_CODING_GENE,
     ]
 
-    assert genome._mutate_output_region(list(dna), 9, rng) is False
-    assert genome._mutate_output_region(list(dna), 10, rng) is True
-    assert genome._mutate_output_region(list(dna), 11, rng) is False
+    assert genome._mutate_output_region(list(dna), 9, rng) is True
+    assert genome._mutate_output_region(list(dna), 10, rng) is False
+    assert genome._mutate_output_region(list(dna), 11, rng) is True
+
+
+def test_correct_number_of_mutations(mutation_rate, rng_seed):
+
+    n_inputs = 2
+    n_outputs = 1
+    n_columns = 10
+    n_rows = 2
+    levels_back = 5
+    primitives = (cgp.Add, cgp.Sub, cgp.Mul, cgp.Div, cgp.ConstantFloat)
+
+    rng = np.random.RandomState(rng_seed)
+    genome = cgp.Genome(n_inputs, n_outputs, n_columns, n_rows, levels_back, primitives)
+    genome.randomize(rng)
+
+    n_mutations = 0
+    genome_new = genome.clone()
+    genome_new.mutate(mutation_rate, rng)
+    for (gene_0, gene_1) in zip(genome.dna, genome_new.dna):
+        if gene_0 != gene_1:
+            n_mutations += 1
```

Comment:
```suggestion
    n_mutations = len([1 for gene0, gene1 in zip(genome.dna, genome_new.dna) if gene0 != gene1])
```

Reply:
You could also import `cgp.genome.count_dna_differences` here.

Reply:
i would not reuse the function from the library, the probability of implementing it wrong two times is trivially smaller than implementing it wrong once ;)

Reply:
> regarding the example: the mutation rate was chosen accordingly to the CGP literature, assuming the implementation is correct, so it should be fine. seems like they all still work, don't they?

I don't know, I haven't run them. The CI only checks whether they crash, but not if they convert to good fitness values.

Reply:
Okay for me.